src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -144,8 +146,43 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 			this._layoutCalledOnce = true;
 		}
 	}
+	reorderGridData(source: IDataResource) {
+		let rowIndex = 0;
+		// Get schema list of grid
+		const columnNames: string[] = source.schema.fields.map(field => field.name);
+		if (source.data.length > 0) {
+			// Check to see if data source is ordered properly based on schema
+			let rowKeys = Object.keys(source.data[0]);
+			if (!equals(columnNames, rowKeys)) {
+				// SQL notebooks does not use columnName as key (instead uses indices)
+				// Indicies indicate the row is ordered properly
+				// We check the data to know if it is in index form or string form
+				for (let index of rowKeys) {
+					// Check to see if first row keys are in index format (numbers)
+					let currentIndex = typeof Number(index);
+					let nextIndex = typeof Number(rowKeys[currentIndex + 1]);
+					if (currentIndex === 'number' && rowKeys[nextIndex] < (rowKeys.length - 1)) {
+						// Check to see if next element is number (confirms table is already ordered)
+						if (!(nextIndex === 'number')) {
+							this._notificationService.error(localize('orderFailed', "Not able to have mix of numbers and string column names"));
+						}
+						// Only reorder data in form of strings (as index ordered tables are already ordered)
+					} else if (typeof (index) === 'string') {
+						// Order each row based on the schema
+						for (let row of source.data) {
+							let reorderedData = {};
+							for (let key of columnNames) {
+								reorderedData[key] = row[key];
+							}
+							rowIndex = source.data.indexOf(row);
```

Comment:
Using foreach here would be easier since that gets you the index directly 

```
const arr = ['one', 'two'];
arr.forEach((value, index) => {
    console.log(`value=${value} index=${index}`)
})
```